### ISSUE TITLE
*Restart* button and kebab menu that replaced drop-down menu is unclear

### ISSUE THREAD

**Issue by AdamWill** [cockpit-project/cockpit#13316]
Cockpit version: 209
OS: Fedora
Page: Front

Prior to Cockpit 209, the 'Power options' menu looked like this:

![powerdrop](https://user-images.githubusercontent.com/916551/71106199-bcc1e200-2173-11ea-81bc-53fb9a43b118.png)

In Cockpit 209, it looks like this:

![powerkebab](https://user-images.githubusercontent.com/916551/71106233-d400cf80-2173-11ea-955e-32a3cbe51ee6.png)

I understand the problem that we're trying to solve here, but I find the new design unclear.

Per the Patternfly docs, the previous element is a [standard dropdown menu](https://www.patternfly.org/v4/documentation/react/components/dropdown). The new approach is listed as ['with kebab'](https://www.patternfly.org/v4/documentation/react/components/dropdown#with-kebab) and is the eighth option in the list. (The name 'kebab menu' was apparently coined in [this tweet](https://twitter.com/lukew/status/590911111358242816?lang=en), which is pretty disparaging...)

It appears the 'kebab menu' element was introduced as part of Google's material design, so it's worth looking at [Google's docs](https://material.io/components/menus/#dropdown-menu) here. (Neither the Patternfly and Bootstrap docs really cover what these elements are for, that I can find, though there's a [Patternfly blog post from 2016](https://blog.patternfly.org/communication/an-eventstask-drawer-thats-handy/) which talks about kebabs a bit.)

In the Google docs examples of kebab menus they're all quite clearly related by design to the UI element they relate to, but in Cockpit 209 the kebab button is not clearly associated with the **Restart** button next to it except by proximity. They're not part of a single clearly-identifiable UI element. The button is a button, and the kebab menu is just sort of floating in space next to it. It wasn't really clear to me that the kebab menu is related to the button, that it gives you other possible actions besides **Restart** (the default choice).

Note in the Google docs that the kebab menu is used to present a set of *possible actions that can be taken on a given element* - in particular, look at the example under **Behaviour**, where we have several items in a list, each with a kebab menu offering 'Preview', 'Share', 'Get Link' etc. These are not 'either or' choices, but actions. They're not alternatives to the list item - they're things you can do *to it*.

Now if you read down to [*Exposed dropdown menu*](https://material.io/components/menus/#exposed-dropdown-menu) and you'll read this:

"Exposed dropdown menus display the currently selected menu item above the menu. They can be used only when a single menu item can be chosen at a time."

above a picture of exactly the type of dropdown Cockpit *used* to use here. So, thinking about this, I can kinda see the problem the kebab button is trying to fix. The idea of a dropdown is that you have one state selected, you click it and select another state, and the button's state (and the actual effect in the app) is updated. That's not *quite* what we're doing here; we actually have a list of possible actions and we want the chosen action to take effect immediately when clicked (IIRC that's what happens, anyway). So I can see why someone thought a dropdown wasn't quite the right thing.

However I don't think a kebab floating in space next to a button is quite right either, looking at all the docs above and other discussions of menu styles around the internet. We seem to be sort of awkwardly between the two scenarios here. I'm afraid I don't have a great solution to propose, but I just wanted to flag up that I found this a bit confusing and provide a bit of research and ask if it can be reconsidered.

Thanks!

**Comment by martinpitt** [comment]
TBH I'm not happy about the kebab either, as it just contains exactly one option. They both bring up almost the same dialog though, so maybe we can just turn this into one "Power..." (or "Shutdown...") button ("..." to clarify that it's opening a dialog -- is that still a thing in PF4?), and select the concrete action there with a radio button?

**Comment by garrett** [comment]
When we made the button plus kebab for the new overview, I didn't see the split button in PF4 (and I looked for it). Apparently, [there is one now](https://www.patternfly.org/v4/documentation/react/components/dropdown#split-button-action).

However, this is a bit different from PF4's suggestion to use the three dots menu.

(FWIW: The "exposed dropdown menu" is a different widget — it even _looks_ different as there's no separation between the action and additional actions — and it would be the wrong pattern to use for a restart / shutdown action.)

Summary: Let's switch back to the split button.
